tritium-processor/build.gradle

```diff
@@ -0,0 +1,23 @@
+apply plugin: 'com.palantir.external-publish-jar'
+
+dependencies {
+    compile project(':tritium-annotations')
+    compile project(':tritium-api')
```

Comment:
```suggestion
    implementation project(':tritium-api')
```

Reply:
I don't believe we can use java-library dependency types within annotation processor implementations, they're not correctly added to the annotationProcessor class path for dependent projects, but it's hard to catch in same-project tests. I've broken several annotation processors this way!

Reply:
interesting, will the `compile` scope work in Gradle 7 (I think tritium is still on 6.9 for now)

Reply:
That’s a good point, I’m not sure.

Reply:
@CRogers or @fawind Any idea about Gradle 7 and annotation processor dependencies?

Reply:
Yes, `compile` and `runtime` configurations got removed with Gradle 7 ([ref](https://docs.gradle.org/current/userguide/upgrading_version_6.html#sec:configuration_removal)). I would have assumed that at least the `api` scope should have similar behavior to `compile` but not very familiar with annotation processor wiring. If this doesn't work, it sounds like something we should potentially flag to Gradle upstream?

Reply:
Ya, I seem to recall that API didn't work, but I could be mistaken. We can take a deeper in the next couple weeks as we'll need to solve it across all our annotation processors.